drivers/ethernet/phy/phy_qualcomm_ar8031.c

```diff
@@ -0,0 +1,521 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * Inspiration from phy_mii.c, which is:
+ * Copyright (c) 2021 IP-Logix Inc.
+ * Copyright 2022 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT qca_ar8031
+
+#include <errno.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/mdio.h>
+#include <zephyr/net/phy.h>
+#include <zephyr/net/mii.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(phy_qc_ar8031, CONFIG_PHY_LOG_LEVEL);
+
+#define AR8031_PHY_ID1           0x004DU
+#define PHY_READID_TIMEOUT_COUNT 1000U
+
+/* PHY Specific Status Register */
+#define PHY_SPECIFIC_STATUS_REG     0x11
+#define SPEC_STATUS_REG_LINK_MASK   (1U << 10)
+#define SPEC_STATUS_REG_DUPLEX_MASK (1U << 13)
+#define PHY_DUPLEX_HALF             (0U << 13)
+#define PHY_DUPLEX_FULL             (1U << 13)
+#define SPEC_STATUS_REG_SPEED_MASK  (0x3U << 14)
+#define PHY_SPEED_10M               (0U << 14)
+#define PHY_SPEED_100M              (1U << 14)
+#define PHY_SPEED_1000M             (2U << 14)
+
+/* The PHY Debug port address register */
+#define PHY_DEBUGPORT_ADDR_REG 0x1DU
+/* The PHY Debug port data register */
+#define PHY_DEBUGPORT_DATA_REG 0x1EU
+
+/* Debug port registers */
+/* Analog Test Control */
+#define PHY_DEBUGPORT_ANALOG_CTRL          0x0
+#define PHY_DEBUGPORT_ANALOG_CTRL_RX_DELAY (1U << 15)
+/* SerDes Test and System Mode Control */
+#define PHY_DEBUGPORT_SD_SM_CTRL           0x5
+#define PHY_DEBUGPORT_SD_SM_CTRL_TX_DELAY  (1U << 8)
+
+/* PHY MMD device address 3 */
+#define PHY_MMD_DEVICEID3              3U
```

Comment:
MMDs are defined in `include/zephyr/net/mdio.h` generically, use those instead. Same applies for `PHY_MMD_DEVICEID7` below.

Reply:
Updated, thanks.

Reply:
you have moved the definitions from the PHY driver to `include/zephyr/net/mii.h`. The request is to use in your PHY driver the defines already present in `include/zephyr/net/mdio.h`. For example, MMDs are defined in https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/mdio.h#L49-L77, MMD generic registers in https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/net/mdio.h#L79-L101, etc, so no need to re-define them in your driver. Other generic registers (not vendor specific) can be in lines below of that same file.

Reply:
Got it, please review again, thanks.